### ISSUE TITLE
Is it possible to render a table with a row selected by default?

### ISSUE THREAD

**Issue by stephenrs** [AllenFang/react-bootstrap-table#16]
Thanks for the great component! I'm using the selectRow feature, but I can't find a way to render a table with a specific row selected by default, or programmatically control which row is selected. 

Am I missing something, or is this a feature request?

**Comment by AllenFang** [comment]
Yep, it's a good feature for row selection, I'll add this feature in recently, but need some times :))

**Comment by stephenrs** [comment]
Great thanks, I'll look forward to it!

**Comment by AllenFang** [comment]
This issue has been solved on <a href="https://github.com/AllenFang/react-bootstrap-table/pull/18">PR#18</a>

**Comment by stephenrs** [comment]
The committed solution for this issue only solves part of the problem for me. In my case, I need to be able to programmatically select a given row, or at a minimum have the component respect new props when it is re-rendered. The current solution only allows setting the default selected row when the component is first mounted, which isn't so flexible. (also see #19 ).

I've made some modifications and have this working in my dev environment - it causes new props to always set the desired row to the selected state.  This seems to be working fine for me, but there might be a better way to do it. Unfortunately, I'm not familiar enough with git to start a pull request (maybe when I have more time), but the change is pretty simple:

This is what the top of TableBody.js looks like in my dev:

```
function TableBody(props) {
    _classCallCheck(this, TableBody);

    _get(Object.getPrototypeOf(TableBody.prototype), "constructor", this).call(this, props);

    this._setSelectedRowByProps(props);

    this._attachRowSelectFunc();
    this.editing = false;
  }

  _inherits(TableBody, _React$Component);

  _createClass(TableBody, {
    componentDidUpdate: {
      value: function componentDidUpdate(prevProps, prevState) {
        this._attachRowSelectFunc();
      }
    },
    componentWillReceiveProps: {
      value: function componentWillReceiveProps(newProps) {
        this._setSelectedRowByProps(newProps);
      }
    },
    _setSelectedRowByProps: {
      value: function _setSelectedItem(props) {
        var selected = props.selectRow.selected

        if (props.selectRow.mode === Const.ROW_SELECT_SINGLE) {
          //if row selection is single, just pick the first item in 'selected'
          selected = selected.length > 0 ? [selected[0]] : [];
        }
        this.state = {
          currEditCell: null,
          selectedRowKey: selected
        };
      }
    },
```

Hope this helps!

**Comment by AllenFang** [comment]
I understand what you means, I'll spend some time to design it.

**Comment by stephenrs** [comment]
Ok great...also a new requirement has emerged on my project, and it would be very convenient to be able un-select a row from an external module without building in another re-render() cycle. Maybe a statics method on BootstrapTable would be useful(?)

**Comment by AllenFang** [comment]
I considered a statics method on some class before but there's some problem. Maybe I will re-think about that.

**Comment by AllenFang** [comment]
hi @stephenrs I've tried a lot of method for this requirement, include static or instance method. But there's no way to complete. Currently, I use <code>componentWillReceiveProps</code> to achieve requirement but you need to assign the new selected value and re-render table, you can see the detail on <a href='https://github.com/AllenFang/react-bootstrap-table/issues/19'>#19</a>

**Comment by AllenFang** [comment]
About the dynamic select or unselect feature, I've released on v0.9.5 :)

**Comment by ab99484** [comment]
@AllenFang  @stephenrs 

I am going to add the selected option in the `selectRowProp`.
As a result, I have made the code below:

`selectRowProp = {
    mode: 'radio',
    bgColor: '#e9ecef',
    clickToSelect: true,
    hideSelectColumn: true,
    selected: this.props.vehicles[0] && [this.props.vehicles[0].vin],
    onSelect: (row) => this.onRowSelect(row),
  }`

As you can see, it will take some time to get props values. So currently selected is the empty or null value.

How can I fix it?
